### DISCUSSION TITLE
Should we support "x" platform?

### DISCUSSION THREAD

**Discussion by Daeraxa** [pulsar-edit/.github#77] [General]
So this kind of relates to https://github.com/pulsar-edit/pulsar/issues/62 but is also just a general talking point on it and useful as a reference point if people start requesting that we support x, y or z package/binary/OS.

This is a list of some of the more common ones and I'll happily edit this to account for any that I've missed. I am *not* saying that these all need to be supported, just giving options to the team and anyone who may want to create and maintain a community package.

I've tried to put this in some kind of order based on types and importance.

The standard ones I think are essential are **bolded**.

## All OSs
- [ ] [Zero Install](https://0install.net/)
- [ ] [pkgx](https://pkgx.dev/pkgs/)

### Universal binaries 
- [x] **AppImage**
- [x] **tar.gz**

### Universal repositories
- [x] [**FlatPak**](https://flathub.org) - [Community supported](https://flathub.org/apps/dev.pulsar_edit.Pulsar) maintained by [@cat-master21](https://github.com/cat-master21)
- [x] [**Snap**](https://snapcraft.io/) - [Community supported](https://snapcraft.io/pulsar-edit) maintained by [@Ottok](https://github.com/ottok)
- [x] [Nix](https://nixos.org/)
    - [x] [Nixpkgs](https://github.com/nixos/nixpkgs) - [Community supported](https://search.nixos.org/packages?channel=unstable&show=pulsar) maintained by [@bryango](https://github.com/bryango) and [@pbsds](https://github.com/pbsds)
    - [ ] [NUR](https://github.com/nix-community/NUR)
- [ ] [GNU/Guix](https://packages.guix.gnu.org/)
- [x] [AM/AppMan](https://portable-linux-apps.github.io/) - [Community supported](https://github.com/ivan-hc/AM-Application-Manager/blob/main/programs/x86_64/pulsar)
- [ ] [Docker](https://hub.docker.com/) (see comment about x11docker: https://github.com/orgs/pulsar-edit/discussions/77#discussioncomment-8003501)

### Distro specific binaries
- [x] **deb** for Debian based distros (Ubuntu etc.)
- [x] **rpm** for RedHat based distros (Fedora etc.)
- [ ] rpm for OpenSUSE based distros (not certain if the "standard" rpm is fully compatible or not)

### Distro specific repositories
- **Debian/Ubuntu**
	- [ ] Official? PPA? Other? (i.e. interact with `apt`)
    - [x] [Pacstall](https://pacstall.dev/) - [Community supported package pulsar-deb](https://pacstall.dev/packages/pulsar-deb) maintained by [@Zahrun](https://github.com/Zahrun) and [@AJStrong](https://github.com/ajstrongdev)
    - [ ] ~~[Makedeb](https://www.makedeb.org/)~~ Makdeb project currently [unmaintained](https://hunterwittenborn.com/blog/stepping-back-from-open-source/)
    - [x] [**deb-get**](https://github.com/wimpysworld/deb-get) - [Officially supported](https://github.com/wimpysworld/deb-get/blob/main/01-main/packages/pulsar)
- **Fedora**
    - [ ] [Fedora packages](https://packages.fedoraproject.org/)
    - [ ] [COPR](https://copr.fedorainfracloud.org/)
    - [ ] [Terrapkg](https://github.com/terrapkg)
- **Arch**
    - [ ] [Official](https://archlinux.org/packages/)
    - [x] [AUR](https://aur.archlinux.org/) - [Community supported package pulsar-bin](https://aur.archlinux.org/packages/pulsar-bin) maintained by [@jonian](https://aur.archlinux.org/account/jonian) - Could we adopt as official?
    - [x] [Unofficial User Repositories](https://wiki.archlinux.org/title/Unofficial_user_repositories) - [Alerque - Pulsar-bin](https://wiki.archlinux.org/title/Unofficial_user_repositories#alerque)
- **OpenSUSE**
    - [ ] [OpenSUSE packages](https://software.opensuse.org/)
    - [ ] [OBS](https://build.opensuse.org/)
- [ ] [Void](https://voidlinux.org/packages/)
- [ ] [KaOS](https://kaosx.us/packages/)
- [ ] [Solus](https://dev.getsol.us/)
- **Gentoo**
    - [ ] [Official](https://gitweb.gentoo.org/)
    - [x] [GURU](https://wiki.gentoo.org/wiki/Project:GURU) - [Community supported package pulsar-bin](https://github.com/
gentoo/guru/tree/master/app-editors/pulsar-bin) maintained by [@zhuyifei1999](https://github.com/zhuyifei1999)
- LiGurOS [x] - [Community supported package pulsar-bin](https://gitlab.com/liguros/liguros-repo/-/tree/develop/app-editors/pulsar-bin) maintained by [@zhuyifei1999](https://github.com/zhuyifei1999)
- [ ] [Alpine Linux](https://pkgs.alpinelinux.org/packages)
- [x] [PCLinuxOS](https://ftp.nluug.nl/pub/os/Linux/distr/pclinuxos/pclinuxos/apt/pclinuxos/64bit/RPMS.x86_64/) - [Community supported](https://ftp.nluug.nl/os/Linux/distr/pclinuxos/pclinuxos/srpms/SRPMS.pclos/)
- **Slackware**
    - [ ] [Slackware packages](https://packages.slackware.com/)
    - [x] [SlackDCE](https://slackware.uk/slackdce/) - [Community supported](https://slackware.uk/slackdce/packages/15.0/x86_64/development/pulsar/) maintained by @Charadon
    - [x] [Slackbuilds](https://slackbuilds.org) - [Community supported](https://slackbuilds.org/repository/15.0/development/pulsar/) maintained by @Charadon

### Other repos
- [ ] [Pi-apps](https://github.com/Botspot/pi-apps)

### Binaries
- [x] **dmg**
- [x] zip
- [ ] ~~pkg~~ See [this discussion comment](https://github.com/orgs/pulsar-edit/discussions/77#discussioncomment-4685790) ~vapor trail (@Meadowsys)

### Repositories
- [x] [**Homebrew**](https://brew.sh/) - [Community supported](https://formulae.brew.sh/cask/pulsar#default)
- [ ] App Store

### Binaries
- [x] **.exe installer**
- [ ] ~~**.exe portable**~~ - Used to but removed as too slow
- [x] **.zip** 
- [ ] .msi installer

### Repositories
- [x] [**Chocolately**](https://chocolatey.org/) - [Officially supported](https://community.chocolatey.org/packages/pulsar)
- [x] [Scoop](https://scoop.sh/) - [Community supported](https://scoop.sh/#/apps?q=pulsar&id=b73e2993681c0efcdf8f5eb26582458e106e7d05)
- [x] [Windows Package Manager aka Winget](https://github.com/microsoft/winget-pkgs) - [Community Supported](https://winstall.app/apps/Pulsar-Edit.Pulsar)
- [ ] Windows store
- [ ] [Portable apps](https://portableapps.com/)

## BSD???

**Comment by confused-Techie** [comment]
I do just want to add. Trying to get these listed on the MacOS App Store, and the Windows Store could be extremely helpful to avoid signing at all. Since once on those stores it's already trusted, and could negate the issue altogether

**Reply by mauricioszabo** [reply]
Actually, it's the opposite - we can ONLY list on these stores if we sign them...

**Reply by confused-Techie** [reply]
> Actually, it's the opposite - we can ONLY list on these stores if we sign them...

Well that's just lame. Guess things had changed since last I checked our read it wrong. Thanks for pointing out the correction

**Reply by meadowsys** [reply]
Hmmmmmmm, I'm wondering if macOS is viable now, since I believe there is no extra cost than the $99/yr to get it listed on the macOS app store?

**Reply by confused-Techie** [reply]
@Meadowsys I'd have no issue with pursuing it at all. As long as an Electron app can be listed properly. I'd think the biggest concern would be file system access. Since I know the Windows store likes to be extra secure about that sort of thing, but am not to sure how it's handled on Mac

**Reply by meadowsys** [reply]
On macOS there is a permission you can give to an app to get full disk access. I wonder if we are able / should request it from Pulsar automatically or if we should just instruct people to add the permission manually

**Comment by Spiker985** [comment]
I would also like to throw the hat into the ring for Windows msi.

Electron-builder can already make one, and it allows for a scriptable install/quiet install

**Reply by Daeraxa** [reply]
Makes sense, I've added it to the list at least.

**Reply by Spiker985** [reply]
I was going to add it as a target on my own branch, and then realized that I would need a pulsar-edit/pulsar branch to actually get Cirrus to build it lol

**Reply by Daeraxa** [reply]
You can still run it manually, if it builds locally then there shouldn't be any reason why Cirrus can't build it.

**Reply by Spiker985** [reply]
I'd have to set up a dev instance on a Windows box, which is doable - but not done yet

But good point, I forget it only runs the `yarn dist` command, in essence

**Reply by confused-Techie** [reply]
@Spiker985 if needed I could submit a PR with any changes, do you have a repo I can look at for the implementation? Or is just like added the other build target like you did previously?

**Reply by confused-Techie** [reply]
So quick update, as I tried to get Cirrus to build an MSI over on [pulsar#279](https://github.com/pulsar-edit/pulsar/pull/279). But it seems MSI's won't work with our version naming. 

Essentially our build number, as it currently uses the date, is a higher value than MSI's will accept, which the maximum number you can specify as an MSI build number (according to Cirrus Error Logs) is 65536. Which we easily go over that.

So this may mean we have to have custom build numbers for an MSI to be made, or otherwise we could find a way to resolve that issue.

**Comment by comcloudway** [comment]
I already tried building pulsar on alpine,
but from what I can tell, `node-keytar` fails to build (assuming I'm not skipping over some important thing)

**Reply by Daeraxa** [reply]
Well this is more about what should we support in terms of packages and repos. It should be buildable on nearly everything so if you need help feel free to ask in the #build channel on discord or open a new thread.

**Comment by meadowsys** [comment]
macOS PKG installers are rare. I have only seen PKG installers used for things like audio plugins, drivers, and this audio loopback thingie I installed.

I usually see DMG installers only (as in, it is the only option provided). Sometimes I see zip files being used (maybe because they are easier to package? but for us DMGs are automated anyways). I also get the impression that DMG files can compress the application bundle down a lot more than ZIP files can.

TL;DR I do not think a PKG installer is needed for macOS. Honestly, offering only a DMG should be enough?

**Reply by Daeraxa** [reply]
I agree personally, just covering all bases in case somebody asks.

**Reply by mauricioszabo** [reply]
Originally, pkg was added by me to allow universal DMG packages (the doc I saw mentioned combining two pkg files). Unfortunately, Cirrus does not have sufficient disk space to build both DMG and pkg for both platforms, and that also made the build slow and brittle.

So, let's drop pkg indeed :+1:

**Comment by Eschguy** [comment]
I'm a big Flatpak advocate, so I'd vote for that.

**Comment by kaosine** [comment]
Definitely need flatpak, but probably support copr as it also supports opensuse and a few other distros if configured right. I was trying to figure this out to add to the existing build config but the documentation for it confused the heck out of me....

**Comment by Digitalone1** [comment]
In my opinion Flatpak is the only thing the team should ship on its own and it covers the most number of Linux distributions.

For other formats, the distributions will do them if they want like the AUR package.

**Reply by Eschguy** [reply]
With Flathub gearing up to support markers for official packages, I think it's a good idea to get on that boat early.

**Reply by kaosine** [reply]
> With Flathub gearing up to support markers for official packages, I think it's a good idea to get on that boat early. 

This, flatpak alone would help us tremendously and maybe snap down the line. That way we're in repos people already have on their systems. I'm still going to eventually look at copr as well but will have to figure things out.

**Comment by Digitalone1** [comment]
To answer the open post, the AUR is made by @jonian 

It's not official because on Arch only the packages on upstream are "official". On AUR they're all user made, but it works.

**Reply by Daeraxa** [reply]
I think "official" in this context refers to "Pulsar-Edit" supported rather than community member. i.e. if we wanted we could have an "official" Pulsar AUR entry with associated pkgbuild within the pulsar-edit repos.

**Reply by Digitalone1** [reply]
> I think "official" in this context refers to "Pulsar-Edit" supported rather than community member. i.e. if we wanted we could have an "official" Pulsar AUR entry with associated pkgbuild within the pulsar-edit repos.

I think it will become outdated because there's no one who update it. The only thing official made here about linux should be the flatpak build in my opinion.

**Comment by anonCoffee** [comment]
**FlatPak** & **Snap** must be supported with regular updates. 

Distro specific support is also welcome.

Can we just stop caring about Windows and Mac users? 
Let's just forget about them.

**Reply by Digitalone1** [reply]
> **FlatPak** & **Snap** must be supported with regular updates.

I'd say only Flatpak since Snap is bad and maybe Canonical in the future will drop it in favor of Flatpak.

> Can we just stop caring about Windows and Mac users? Let's just forget about them.

No.

**Reply by xxorax** [reply]
> Snap is bad and maybe Canonical in the future will drop it in favor of Flatpak

Source ? Snap is highly committed on the last Ubuntu.

**Reply by anonCoffee** [reply]
What's the use of supporting Windows & Mac platforms officially?

**Reply by anonCoffee** [reply]
Through these discussions, Github is tracking me shamelessly. What's the use of Github? Why not migrate to a better alternative? Proprietary software is a pain, will always remain a pain. Rest, whatever you guys decide.

**Reply by Digitalone1** [reply]
> Source ? Snap is highly committed on the last Ubuntu.

If I have to trust a company, I'd choose Red Hat over Canonical. I think Flatpak will be the standard in the next years. Maybe I will be wrong, but at the moment I recommend Flatpak. The history does not speak good about Canonical, look what happened to Unity Desktop. 

> What's the use of supporting Windows & Mac platforms officially?

Because they have almost 99% market share?

> Through these discussions, Github is tracking me shamelessly. What's the use of Github? Why not migrate to a better alternative? Proprietary software is a pain, will always remain a pain. Rest, whatever you guys decide.

It works, so we stay on it at the moment. In the future the plan is to migrate, but we're not in a hurry right now.

**Reply by Daeraxa** [reply]
> What's the use of supporting Windows & Mac platforms officially?

Other than one of our principal contributors is a Windows user? I feel that should be enough.

> Through these discussions, Github is tracking me shamelessly. What's the use of Github? Why not migrate to a better alternative? 

We have discussed this endlessly, basically Pulsar is still very much entwined with GitHub as it is (although we are working to undo things). We also want to get as much traction on the project as possible with the smallest barrier of entry and GitHub simply has a much larger user base, has much more exposure to the world and is more likely to entice people in. It also allows us to use the GH Discussions area as a forum with zero configuration needed unlike setting up Discourse etc.  
This isn't to say we might not move eventually but it would just be counter productive and a lot of work for no real benefit right now.

I should add that I do think we should be supporting snap. So long as it doesn't add a significant maintenance burden and can be automated then I see no reason to not publish there.

**Reply by confused-Techie** [reply]
@Daeraxa Beautifully put

**Reply by anonCoffee** [reply]
> Other than one of our principal contributors is a Windows user? I feel that should be enough.

Is that a reason? 
It's dumb. 
First, Those who use Microsoft Windows are plain dumb and nothing more. 
Second, Atom was one of the most beautiful FOSS code editor. Pulsar is a great initiative. I hope that it will remain FOSS. 

> Because they have almost 99% market share?

Do you care about FOSS or market share? What is your primary concern?
Microsoft GitHub simply dumped Atom. So that, Pulsar should get the market share? Absolutely not.
As they declared, GitHub dumped it "in order to prioritize technologies that enable the future of software development", specifically its Github Codespaces and Microsoft's Visual Studio Code.

So you guys think that Pulsar is going to get that 99% market share or Codespaces and Microsoft Visual Studio Code?
Anyway. Your project, do whatever. 

I have nothing else to say.

**Reply by confused-Techie** [reply]
@anonCoffee 
Of course Pulsar will remain a FOSS program.

Even if you disagree with the company Microsoft, what advantage do you see dumping Windows and MacOS users will give us? We already have the process and tools to build binaries for these users, it doesn't supply an technical debt to support them, nor do we really need to do anything on our part. It would make very little sense for us to abandon the majority of users of Pulsar based on the idea "Those who use Microsoft Windows are plain dumb and nothing more.".

Pulsar is under the initiative to let everyone contribute, and to lower the barrier to entry for as many users/contributors as we can. Purposefully, and with zero gain, if we were to stop supporting the majority of our users/contributors just doesn't make any sense.

We don't think we are going to get 99% of the market share (Of Editors?), @Digitalone1 was simply saying that Microsoft and MacOS have 99% market share in operating systems. So if we were to stop supporting those platforms we would lose 99% of possible contributors, and people who may love the editor.

Trust us, we aren't fans of Microsoft has done to Atom. That is the entire reason we are here, but there's a difference in not aligning with the values of a company and taking it out on it's users, which again would serve us absolutely zero gain, and would in all likelihood damage the project as a whole.

**Reply by anonCoffee** [reply]
I understand everything that you say. 

> So if we were to stop supporting those platforms we would lose 99% of possible contributors. 

Contributors who do not care about FOSS are not really contributors of FOSS projects. You cannot rely upon them for anything. 


> damage the project as a whole

That's the last thing I would ever wish for. 


> We already have the process and tools to build binaries for these users, it doesn't supply an technical debt to support them, nor do we really need to do anything on our part.

That's right. I didn't exactly mean to prevent them from using it or contributing to it. I meant they cannot be the priority or something of much value to the project. 


> Microsoft and MacOS have 99% market share in operating systems. So if we were to stop supporting those platforms we would lose 99% of possible contributors, and people who may love the editor.

That's the dangerous thing. 


The dangerous thing is that a FOSS project is going to die if users of proprietary software don't care about it or don't get access to it. My point is not banning or prevent them from using it. My point is dependence on them. So, a FOSS is dependent on a proprietary software to exist indirectly? 
I said stop caring about them. I didn't mean ban them. 

One more thing, for donations, add crypto as an option too. 

This was my opinion. 

Take it or leave it. 

I would love to see FOSS Pulsar outgrow any other editor, VSCode, Codespaces or any other proprietary headache.

**Reply by Digitalone1** [reply]
> First, Those who use Microsoft Windows are plain dumb and nothing more. 

I would ban you from here only for saying this. Respect other people choice. The discussion is over for me.

**Reply by anonCoffee** [reply]
> I would ban you from here only for saying this. Respect other people choice. The discussion is over for me.

Ban me. 

I have zero respect for Microsoft users and admirers. No question of respecting idiots. 
Microsoft users are plain dumb and nothing more. 

Go ahead. Ban me.

**Reply by Daeraxa** [reply]
> I have zero respect for Microsoft users and admirers. No question of respecting idiots. Microsoft users are plain dumb and nothing more.

Your thinking here is really rediculous. I use Linux at home but what about work? I don't have any choice other than to use Windows and why would I not want to use the editor I'm spending time trying to improve? Not every Windows user is an "admirer".

**Reply by anonCoffee** [reply]
> I don't have any choice 

That's my point. 

Anyway, I think it's stretching more than necessary. So, let's leave it here. 
I am just ending this Windows point here. Whosoever uses whatsoever software, not my concern. 
I am here for Pulsar. 


> The editor I'm spending time trying to improve.

Great initiative. Great project. Keep it up. 
I am happy to give my life and blood for this project as well. 

Let's hope that Pulsar takes all of us far ahead. 

I am happy to meet you guys. 


Have a nice day!

**Reply by Daeraxa** [reply]
Thanks for the support, ultimately we just want to support the application and users so unfortunately that does mean making sacrifices in terms of ideals sometimes (for example, do I support GitHub? No, but the benefits it gives to the application outweighs the idealistic negatives for the time being).

**Comment by MrUnkert** [comment]
As a high school teacher I really like the .deb support as I can install this excellent, excellent editor on my Chromebook! Thank you so much for putting energy into this project!

**Reply by confused-Techie** [reply]
Thanks a ton for your kind words. Super happy to hear this provides some value to you!

**Comment by miccal** [comment]
https://github.com/Homebrew/homebrew-cask/pull/141442

**Comment by COLAMAroro** [comment]
Discovered this discussion late, but I've been working on Nix, via the nixpkgs

https://github.com/NixOS/nixpkgs/pull/221862

**Reply by COLAMAroro** [reply]
PR got accepted, and it now has been backported to all major channels !
https://nixpk.gs/pr-tracker.html?pr=221862

Pulsar is now officially available in the nixpkgs :)
(ping @Daeraxa)

**Reply by Daeraxa** [reply]
Awesome, not sure how to link to the package itself on the site other than via a query but I'll add it to the list - https://search.nixos.org/packages?channel=unstable&show=pulsar

**Reply by COLAMAroro** [reply]
Either use the query, or the actual recipe:
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/editors/pulsar/default.nix#L152

But both are good :) (there is no standard for this, so both are good. Even then with the query, the button "show source" links back to the nix recipe)

**Reply by Daeraxa** [reply]
Definitely want a link to the site as that is far more what end users would expect but it seems really weird that each package doesn't actually have a URL on the website for purpose of linking it to other people.

**Reply by COLAMAroro** [reply]
I wish there was something like this, but alas the Nix documentation is still lacking...

**Reply by COLAMAroro** [reply]
# How to update the package

(This guide is meant for people who want to also to contribute to keeping this package up to date and become a maintainer)

## Basics

You can probably just do it through the usual [contributing guide from Nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
As you might guess, you will need Nix installed to actually try to build and run the package.

If nothing major changes in how the package is published, you could probably just run the `$NIXPKGS_PATH/pkgs/applications/editors/pulsar/update.mjs`, which will update the hashes and the paths to the correct tarball.

After that, it's the usual `nix-build -A pulsar` from the root of your Nixpkgs clone to see if this build correctly, and `./result/bin/pulsar` to start it.

If it's good, it's the usual PR on GitHub :)

## Some gotchas

### aarch64

The derivation also supports aarch64. Cross-building is having issues so you could try it but I'm not assured it would work. I would advise instead to try to build from a free Oracle Ampere VM, or a raspberry Pi. I'm not sure if it's *required* to test if it builds on aarch64, but it's a nice move if you do

### Change in the release format

The update script assumes that release have a special formats on GitHub:

1. A file called `SHA256SUMS.txt`, with all the hashes in a format `SHA256HASH Filename`
  1.1. Example file here: https://github.com/pulsar-edit/pulsar/releases/download/v1.103.0/SHA256SUMS.txt
2. The x86_64 Linux file should be called `Linux.pulsar-${VERSION}.tar.gz`
3. The ARM Linux file should be called `ARM.Linux.pulsar-${VERSION}-arm64.tar.gz`

If those file name changes, the update script will fail. In that case you will probably need to update the file name  in the Nix file. (If you can, also update the updater script)

### Change in the build

The Pulsar package is based on the prebuilt tarball. If it's format changes from version to version, the patches needed to make it work on Nix will break.
For example, the ELF of the prebuilt Pulsar [needs a patch to replace some libraries](https://github.com/NixOS/nixpkgs/blob/aa4cf231c4d1d062dd5e91d5d7807da4442c4f5f/pkgs/applications/editors/pulsar/default.nix#L92-L99).
If the Pulsar binary changes place or requires new library, the current patch will not suffice.

Another sort of brittle part is the [wrapper script](https://github.com/pulsar-edit/pulsar/blob/fe89387509f548716976ad75f94bb5578db3f4ce/pulsar.sh#L4). It assumes that the Pulsar bin is at `/opt/Pulsar/pulsar` which is ***not*** where it is going to end-up when managed by Nix. In our case, we apply [a patch](https://github.com/NixOS/nixpkgs/blob/aa4cf231c4d1d062dd5e91d5d7807da4442c4f5f/pkgs/applications/editors/pulsar/001-patch-wrapper.patch) that replace that assumptions with an environment variable (which will be filled later down the line by another part of the build)

**Comment by confused-Techie** [comment]
We've got a plus one for scoop [`pulsar-edit/pulsar#447`](https://github.com/pulsar-edit/pulsar/issues/447)

**Comment by COLAMAroro** [comment]
Wrote a Scoop JSON. It probably can get enhanced (like automated new version discovery), but at least that's a great first step :)

```json
    {
        "version": "1.103.0",
        "license": "MIT",
        "architecture": {
            "64bit": {
                "url": "https://github.com/pulsar-edit/pulsar/releases/download/v1.103.0/Windows.Pulsar.1.103.0.exe",
                "hash": "sha256:40872f766c5d758add4cab9a4d8ace37f07dadd4ee5240ea81d7c096a169f0c2"
            }
        },
        "pre_install": "Rename-Item \"$dir\\Windows.Pulsar.$version.exe\" \"pulsar.exe\"",
        "bin": "pulsar.exe",
        "homepage": "https://pulsar-edit.dev/",
        "description": "A Community-led Hyper-Hackable Text Editor",
        "autoupdate": {
            "architecture": {
                "64bit": {
                    "url": "https://github.com/pulsar-edit/pulsar/releases/download/v$cleanVersion/Windows.Pulsar.$cleanVersion.exe"
                }
            }
        }
    }
```

**Comment by Fargotroniac** [comment]
Hello,

Do you plan to officially support snap?

**Reply by Daeraxa** [reply]
There is no official standpoint on it yet but I would personally say yes we should (and why it is bolded in the post).

**Comment by bx9874** [comment]
Great project! I wish you lots of success.
I don't know what your plans are, but it would be really great if you could also provide a Pulsar Docker image for x11docker.
If that is not possible, Pulsar would certainly be a real asset to Alpine Linux.
